tools/topo_effect_kcontrols.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+"""Parses the .tplg file argument and returns a list of effect
+kcontrols of BYTES type, one per line.
+
+Pro tip: try using these commands _interactively_ with ipython3
+"""
+
+# Keep this script short and simple. If you want to get something else
+# from .tplg files, create another script.
+
+import sys
+from tplgtool2 import TplgBinaryFormat, TplgType, DapmType, SofVendorToken
+
+TPLG_FORMAT = TplgBinaryFormat()
+
+def main():
+    "Main"
+
+    parsed_tplg = TPLG_FORMAT.parse_file(sys.argv[1])
+    component = sys.argv[2]
+
+    # pylint: disable=invalid-name
+    DAPMs = [
+        item for item in parsed_tplg if item.header.type == TplgType.DAPM_WIDGET.name
+    ]
+
+    for dapm in DAPMs:
+        effect_blocks = [b for b in dapm.blocks if b.widget.id == DapmType.EFFECT.name]
+
+        for gb in effect_blocks:
+            # debug
+            # print(f"{gb.widget.id}: {gb.widget.name}")
+            if gb.widget.name == component:
+                print_bytes_kcontrols(gb)
+
+
+def print_bytes_kcontrols(effect_block):
+    "Print bytes kcontrols"
+
+    bytes_kcontrols = [
+        kc
+        for kc in effect_block.kcontrols
+        if kc.hdr.type == 'BYTES'
+    ]
+
+    wname_prefix = (
+        f"{effect_block.widget.name} " if has_wname_prefix(effect_block.widget) else ""
+    )
+
+    for vkc in bytes_kcontrols:
+        print(wname_prefix + vkc.hdr.name)
+
+
+# This could probably be moved to tplgtool2.py?
+def has_wname_prefix(widget):
+    """Is the kcontrol name prefixed with the widget name? ("PGAxx" or "Dmicxx")
+    Check SOF_TKN_COMP_NO_WNAME_IN_KCONTROL_NAME"""
```

Comment:
That's a lot of duplication with `topo_vol_kcontrols.py`. Please you find some place for this in the already shared `tplgtool2.py` and move it there. That shouldn't be hard.

I will make this script 1/3 shorter and the other one 1/2 shorter!

EDIT: just realized I left this exact TODO in `topo_vol_kcontrols.py`!

Reply:
Done, and seems to work though not sure I did it right. I'm not familiar with python classes that most of the stuff in tplgtool2.py seems to be.

Reply:
I don't know whether there is a better place or better way either, but at least not it's not duplicated and easy to make better if needed. Thanks!